### ISSUE TITLE
[BUG] Attachments sometimes fail to render on initial loads.

### ISSUE THREAD

**Issue by Spiral-Memory** [RocketChat/EmbeddedChat#533]
### Description:

Attachments don't render properly on the first deployment without visiting the RC app first.

### Steps to reproduce:

1. Deploy Embedded Chat.
2. Immediately send multiple different attachments such as photos/videos, etc.
3. Notice that they won't render properly.

### Expected behavior:

Attachments should render properly.

### Actual behavior:

Attachments render only when you first visit Rocket Chat and then refresh the EC app.

Note: I'm unsure whether this issue is related to Embedded Chat or Rocket Chat. Also, the issue is hard to reproduce, so not attaching a video.

**Comment by JeffreytheCoder** [comment]
@Spiral-Memory Great find! I had this issue for a long time...any idea why the files won't render?

**Comment by Spiral-Memory** [comment]
Nope.. but if you just open rc once and refresh ec. It will start to render.. 

Why it happens is unknown !

**Comment by JeffreytheCoder** [comment]
> Nope.. but if you just open rc once and refresh ec. It will start to render..
> 
> Why it happens is unknown !

I did this but it still doesn't render on my side

**Comment by Spiral-Memory** [comment]
Sometimes, that happens too 😂 but it mostly works..

**Comment by Spiral-Memory** [comment]
The exact cause is unknown !

**Comment by JeffreytheCoder** [comment]
I found out the issue - I disabled the **Settings -> File Upload -> Protect Uploaded Files** and the files render perfectly :)

![file-not-rendered](https://github.com/RocketChat/EmbeddedChat/assets/35394596/41069d61-041f-441b-8190-3fb562767f23)

I'm updating the Readme so that EC contributors are aware of this

**Comment by Spiral-Memory** [comment]
Hey, but why ? Why will i disable the option to protect uploaded files ? 

Let's not keep it in readme.. it's a vague solution with no explanation involved.

**Comment by JeffreytheCoder** [comment]
Check out this existing [issue](https://github.com/RocketChat/Rocket.Chat.Electron/issues/227) in RC-Electron. 

Basically, the download link is opened in the default browser, not in the RC client (EC in our case). This is why in your case, you must log into RC in the browser before being authenticated to see the files. 

Looks like this is an unsolved issue since 2016...so disabling the file protection option is a workaround for now

**Comment by Spiral-Memory** [comment]
Great find, in my opinion, let's keep this issue and solution here.. or maybe I'll ask to include it in the FAQ.. Keeping it in readme won't be good in my opinion !

**Comment by JeffreytheCoder** [comment]
well it's better to let more people know about this issue and the workaround...putting this on the FAQ is a good idea

**Comment by Spiral-Memory** [comment]
Awesome !

**Comment by sidmohanty11** [comment]
Good find @JeffreytheCoder !

**Comment by Spiral-Memory** [comment]
Closing due to inactivity.
